src/contracts/libraries/BN254.sol

```diff
@@ -260,11 +307,18 @@ library BN254 {
         return keccak256(abi.encodePacked(pk.X, pk.Y));
     }
 
+    /// @return the keccak256 hash of the G2 Point
+    /// @dev used for BLS signatures
+    function hashG2Point(
+        BN254.G2Point memory pk
+    ) internal pure returns (bytes32) {
+        return keccak256(abi.encodePacked(pk.X[0], pk.X[1], pk.Y[0], pk.Y[1]));
+    }
 
     /**
      * @notice adapted from https://github.com/HarryR/solcrypto/blob/master/contracts/altbn128.sol
      */
-    function hashToG1(bytes32 _x) internal view returns (uint256, uint256) {
+    function hashToG1(bytes32 _x) internal view returns (G1Point memory) {
```

Comment:
discussed offline with @gpsanant -- we can remove the comment below (`// XXX: Gen Order (n) or Field Order (p) ?`) carried over from the code this function is adapted from; the use of the field order is correct here.

Reply:
resolved https://github.com/Layr-Labs/eigenlayer-contracts/pull/35/commits/056c33ebe34279a5c4a7e66e81554d289674994d